ggml/src/ggml-sycl/element_wise.cpp

```diff
@@ -32,66 +34,72 @@ static void gelu_f32(const float * x, float * dst, const int k,
     }
 
     float xi = x[i];
-    dst[i] = 0.5f * xi *
-             (1.0f +
-              sycl::tanh(SQRT_2_OVER_PI * xi * (1.0f + GELU_COEF_A * xi * xi)));
+    dst[i] = static_cast<T>(0.5f) * xi *
```

Comment:
Why need to update here?

dst[i] and xi are float type.
So convert 0.5f to T will lead to same result of legacy code.

Reply:
`dst[i]` and xi are "T" type here and not float type. T can be both float and half depending upon the input since I am using a templated function here.

Reply:
Yes! I miss to see the new type of them.
It's OK!